libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridSplineToGrid.cpp

```diff
@@ -0,0 +1,904 @@
+{
+    Point intersectionPoint;
+
+    // Need way of getting the number of times the spline intersect.
+    if (splines.GetSplinesIntersection(splineI, splineJ, crossProductOfIntersection, intersectionPoint, firstNormalisedIntersectionLength, secondNormalisedIntersectionLength))
+    {
+        numberTimesCrossing = 1;
+    }
+    else
+    {
+        secondNormalisedIntersectionLength = constants::missing::doubleValue;
+        firstNormalisedIntersectionLength = constants::missing::doubleValue;
+        crossProductOfIntersection = constants::missing::doubleValue;
+        numberTimesCrossing = 0;
+    }
+}
+
+void meshkernel::CurvilinearGridSplineToGrid::DoubleSplinePoints(Splines& splines) const
+{
+    std::vector<Point> doubledSplinePoints;
+    doubledSplinePoints.reserve(2 * splines.Size(splines.MaxSizeIndex()) - 1);
+
+    for (UInt splineIndex = 0; splineIndex < splines.GetNumSplines(); ++splineIndex)
+    {
+        doubledSplinePoints.resize(2 * splines.Size(splineIndex) - 1);
+
+        // Copy original spline points to every second point
+        for (UInt j = 0; j < splines.Size(splineIndex); ++j)
+        {
+            doubledSplinePoints[2 * j] = splines.m_splineNodes[splineIndex][j];
+        }
+
+        // Evaluate the spline at the intermediate points
+        for (UInt j = 1; j < splines.Size(splineIndex); ++j)
+        {
+            double segmentMidPoint = static_cast<double>(j) - 0.5;
+            doubledSplinePoints[2 * j - 1] = splines.Evaluate(splineIndex, segmentMidPoint);
+        }
+
+        // Replace the spline points with the new double size point set.
+        splines.Replace(splineIndex, doubledSplinePoints);
+    }
+}
+
+bool meshkernel::CurvilinearGridSplineToGrid::ComputeAndCheckIntersection(Splines& splines,
+                                                                          const UInt splineI,
+                                                                          const UInt splineJ,
+                                                                          std::vector<int>& splineType,
+                                                                          VectorOfDoubleVectors& splineIntersections) const
+{
+    UInt maxSplineSize = splines.Size(splines.MaxSizeIndex());
+
+    double crossProduct;
+    double normalisedIntersectionSplineI;
+    double normalisedIntersectionSplineJ;
+    UInt numberTimesCrossing = 0;
+
+    DetermineIntersection(splines, splineI, splineJ, numberTimesCrossing, crossProduct, normalisedIntersectionSplineI, normalisedIntersectionSplineJ);
+
+    if (numberTimesCrossing == 1)
+    {
+
+        if (splineType[splineI] * splineType[splineJ] == 1)
+        {
+            if (maxSplineSize > MaximumNumberOfSplinePoints / 2)
+            {
+                throw AlgorithmError("splines both in m- and n-direction: spline {} and {}", splineI + 1, splineJ + 1);
+            }
+            else
```

Comment:
redundant else

        if (splineType[splineI] * splineType[splineJ] == 1)
        {
            if (maxSplineSize > MaximumNumberOfSplinePoints / 2)
            {
                throw AlgorithmError("splines both in m- and n-direction: spline {} and {}", splineI + 1, splineJ + 1);
            }
            return true;
        }
        if (splineType[splineI] == 0 && splineType[splineJ] == 0)
        {
            throw AlgorithmError("Both spline {} and {} are undefined", splineI + 1, splineJ + 1);
        }